blocks/flag.ts

```diff
@@ -75,12 +76,33 @@ const flagBlock: Block<BlockModule<FlagFunc>> = {
       },
     };
     if (isMultivariate(flag)) {
-      const value = ((flag?.variants ?? []).find((variant) =>
-        typeof variant?.rule === "function" && variant?.rule(ctx)
-      ) as Variant<unknown>)?.value ??
-        ((flag?.variants ?? [])[flag?.variants?.length - 1] as Variant<unknown>)
-          ?.value;
-      return isDeferred(value) ? value() : value;
+      let match: Variant<unknown> | null = null;
+
+      for (const variant of flag.variants || []) {
+        if (typeof variant?.rule !== "function") {
+          continue;
+        }
+        let ruleResult = variant?.rule(ctx);
+        // the rule can sometimes be a promise and we need to await it to check if it's truthy or not
+        if (isAwaitable(ruleResult)) {
+          try {
+            ruleResult = await ruleResult;
+          } catch (_) {
+            // if the rule throws an error, we don't match this variant
```

Comment:
Why are you ignoring the errors? shouldn't errors be thrown? couldn't this silently fail in production causing non-debuggable bugs?

Reply:
yeah, reading this code now it doesn't make sense at all, now it will throw if that's the case